### PR TITLE
Improve nap unit tests

### PR DESCRIPTION
**Pull request by pleshakov**
### Proposed changes
Add unit tests/improve existing ones for AppProtect related code

### CODE REVIEW

File: internal/k8s/handlers.go [hunk 1]
```diff
@@ -523,6 +523,7 @@ func createAppProtectPolicyHandlers(lbc *LoadBalancerController) cache.ResourceE
 	return handlers
 }
 
+// compareSpecs returns true if the resources are different.
 func compareSpecs(oldresource, resource *unstructured.Unstructured) (bool, error) {
```
Comment (soneillf5) [inline internal/k8s/handlers.go:527]: Would it help to rename the function "areResourcesDifferent" ?

**Review by soneillf5 — APPROVED** [review]
Great change! There's a lot of very clear tests.

**Review by rafwegv — APPROVED** [review]
Nice work!

File: internal/k8s/handlers.go (hunk 1 above)
Comment (pleshakov) [reply internal/k8s/handlers.go]: good suggestion

**Pull request merged**